### DISCUSSION TITLE
Blog: Specifying State Machines with Temporal Logic

### DISCUSSION THREAD

**Discussion by owickstrom** [quickstrom/quickstrom#101] [Show and tell]
I've just published a blog post: https://wickstrom.tech/programming/2021/05/03/specifying-state-machines-with-temporal-logic.html

Comments, questions, ideas? Let's discuss!

**Comment by fikse** [comment]
@owickstrom Hey thanks for you blog post! I understand the concepts, but I had to ignore the visualizations. I was unable to follow any of them starting with `always`:
```
always P   ●───●───●───●───●
P          ●───●───●───●───●
always Q   ○───○───●───●───●
Q          ●───○───●───●───●
```
I'm not really sure if there's much more you can say, though, than what you have in your blog post, but that's my comment 🙂 Thank you so much for this!

**Reply by owickstrom** [reply]
I see. There's been some feedback about the diagrams being hard to understand. I'm currently hacking a bit on an interactive visualizer with the same style, but where you can toggle the validity of atomic propositions (e.g. `P` and `Q`) in each state and see how that affects various formulae. Maybe that could help with understanding?

**Reply by avidrucker** [reply]
As for the final symbol, how about an arrow or terminating symbol such as a box with a diagonal line running through it? Or an ellipses?

**Reply by avidrucker** [reply]
@owickstrom  Any updates on the visualization? I'm super down to brainstorm and/or assist in research too as it might be useful / helpful ~

**Comment by avidrucker** [comment]
Why use the word "formula" instead of "predicate"?
> a logical expression that evaluates to true or false

**Comment by avidrucker** [comment]
Suggestions:
1. How about making extra clear specific examples with numbered diagrams? So, in this case, you could simply label each image you have made.
2. Or, going further with the concept, you could highlight named, numbered, and/or labelled "transition diagrams" that show the evolution of a model being built, or show the branching paths (or loops) of a state machine. I like to use Figma for this sort of diagram drawing for it's flexibility, and ability to create interactive prototypes. ( On a related note: Not sure if this still works: https://github.com/ddanielbee/figma-fsm )
3. Thirdly, simply draw out one or more FSMs (or pieces of them) that will help readers quickly visually comprehend the message you are conveying through text, for example: https://images.app.goo.gl/MR5mV6iUmkHxn1b38
